### DISCUSSION TITLE
Why does performing inference consumes VRAM?

### DISCUSSION THREAD

**Discussion by malfonsoarquimea** [Lightning-AI/pytorch-lightning#12090] [General]
Once the model is already loaded into memory and so are the data points that will be used to train the model, why performing the forward pass consumes memory? How to calculate how much memory will be consumed?
Thanks very much, I am starting to dig into memory consumption optimization and I have a lot of doubts. If you consider that I should read something to better understand my problem, just let me know. Thanks

**Comment by rohitgr7** [comment]
it requires memory to compute activations and intermediate states when you do a forward pass.

**Reply by malfonsoarquimea** [reply]
Thanks, this is what i wanted to know!
